### PR TITLE
Implement neuroprobe_2025 pipeline, Neuroprobe2025 Dataset

### PR DESCRIPTION
**Pull request by ShivashriganeshMahato**
## Summary by CodeRabbit

* **New Features**
  * Added the Neuroprobe2025 dataset with dual-mode selection, deterministic split handling, and user-facing configuration options
  * Added an end-to-end NeuroProbe 2025 processing pipeline for manifesting, downloading, and producing HDF5 outputs

* **Bug Fixes**
  * Ensure log output directories are created before redirecting worker output

* **Tests**
  * Added comprehensive tests covering dataset selection, metadata, and edge cases

### CODE REVIEW

File: brainsets_pipelines/neuroprobe_2025/pipeline.py [hunk 1]
```diff
@@ -2,69 +2,548 @@
...
+                _data_set_nested_attribute(data.splits, split_key, intervals)
+
+        # save data to disk
+        self.update_status("Storing")
+        path = self.processed_dir / Path(fpath).name
+        with h5py.File(path, "w") as file:
+            data.to_hdf5(file, serialize_fn_map=serialize_fn_map)
+        logging.info(f"Saved data to {path}")
+    
+    def iterate_extract_splits(self, subject_id: int, trial_id: int) -> Tuple[Dict, ArrayDict]:
```
Comment (glchau) [inline brainsets_pipelines/neuroprobe_2025/pipeline.py]: This will fail in the --single loading setup because we only download a single session while split generation requires other sessions. We could add a warning and skip cross-session / subject if not available

File: brainsets_pipelines/neuroprobe_2025/pipeline.py [hunk 2]
```diff
@@ -0,0 +1,627 @@
...
+            dtype=dtype,
+            lite=lite,
+            nano=nano,  # NOTE: We add nano here for completeness
+            binary_tasks=binary_tasks,
+            output_indices=True,
+            output_dict=True,
+            start_neural_data_before_word_onset=start_neural_data_before_word_onset,
+            end_neural_data_after_word_onset=end_neural_data_after_word_onset,
+            max_samples=max_samples,
+        )
```
[the remark this answers is not in the document]
Comment (glchau) [reply brainsets_pipelines/neuroprobe_2025/pipeline.py]: This leakage should not happen because Neuroprobe2025 dataset will return a different subject/session to be used as training than for test https://github.com/neuro-galaxy/brainsets/pull/79/changes#diff-1f08fb05671ecbbfa219dd84b08750599725567a3937fb8ed75149cd7ff10d91R538

File: brainsets/datasets/Neuroprobe2025.py [hunk 3]
```diff
@@ -0,0 +1,608 @@
...
+                    f"for subject {self.test_subject}, found {longest_trials}."
+                )
+            if self.test_session not in longest_trials:
+                raise ValueError(
+                    "SS-DM full benchmark-default only supports target sessions present "
+                    f"in NEUROPROBE_LONGEST_TRIALS_FOR_SUBJECT for subject {self.test_subject}: "
+                    f"{longest_trials}."
+                )
+
+    @classmethod
```
[the remark this answers is not in the document]
Comment (glchau) [reply brainsets/datasets/Neuroprobe2025.py]: not really, I'll change it to a private instance method!

[the remark this answers is not in the document]
Comment (glchau) [reply brainsets/datasets/Neuroprobe2025.py]: actually, incorporated it into validate_split_args and made a public class method

File: tests/test_neuroprobe2025_dataset.py [hunk 4]
```diff
@@ -0,0 +1,542 @@
...
+import h5py
+import numpy as np
+import pytest
+from temporaldata import Interval
+
+from brainsets.datasets.Neuroprobe2025 import (
+    Neuroprobe2025,
+    _from_recording_id,
+    _to_recording_id,
+)
```
[the remark this answers is not in the document]
Comment (glchau) [reply tests/test_neuroprobe2025_dataset.py]: fyi the mixin has been renamed

File: brainsets/pipeline.py [hunk 5]
```diff
@@ -176,6 +176,9 @@ def _run_item_on_parallel_worker(self, manifest_item):
         log_out_path = log_dir / f"{self._asset_id}.out"
         log_err_path = log_dir / f"{self._asset_id}.err"
 
+        ## self._asset_id may be nested folder path, ensure parent dirs exist
+        log_out_path.parent.mkdir(parents=True, exist_ok=True)
+
         with _redirect_stdio(log_out_path, log_err_path):
```
Comment (vinamarora8) [inline brainsets/pipeline.py]: Do we need this here? The `pipeline_logs` dir seems flat when I run the pipeline

Comment (glchau) [reply brainsets/pipeline.py]: You're right it's flat for neuroprobe_2025. No reason to add. Will remove.

File: brainsets_pipelines/neuroprobe_2025/pipeline.py [hunk 6]
```diff
@@ -0,0 +1,674 @@
+# /// brainset-pipeline
+# python-version = "3.11"
+# dependencies = [
+#   "neuroprobe==0.1.7",
```
[the remark this answers is not in the document]
Comment (glchau) [reply brainsets_pipelines/neuroprobe_2025/pipeline.py]: Yea unfortunately neuroprobe requires pytorch, so until that gets adjusted we'll have to keep importing all its dependencies.

File: CHANGELOG.md [hunk 7]
```diff
@@ -9,6 +9,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Added generic data extraction helpers in `mne_utils` to handle MNE Raw objects ([#78](https://github.com/neuro-galaxy/brainsets/pull/78)).
 - Enriched `s3_utils` with additional functionalities to get data from public buckets ([#78](https://github.com/neuro-galaxy/brainsets/pull/78)).
 - Added new utility functions to split data into train/valid/test splits intersession, intrassession, and intersubject ([#83](https://github.com/neuro-galaxy/brainsets/pull/83)).
+- Added the `Neuroprobe2025` dataset class and the `neuroprobe_2025` brainset pipeline.
```
Comment (vinamarora8) [inline CHANGELOG.md]:
```suggestion
- Added the `Neuroprobe2025` dataset class and the `neuroprobe_2025` brainset pipeline ([#79](https://github.com/neuro-galaxy/brainsets/pull/79)).
```

**Review by vinamarora8 — APPROVED** [review]
Thanks for all the changes and being patient with the back-and-forth! Good to merge now.
I think you might also have to accept it to clear the "Changes Requested"
